### ISSUE TITLE
[PROJECT] Hacktoberfest '21

### ISSUE THREAD

**Issue by BenJam** [opencollective/opencollective#4595]
## Overview
As part of Hacktoberfest this year folks will be able to contribute $ to OS projects as well as code. Users to donate>$20 get a Hacktoberfest tee shirt. 

## Owner
Lead: @benjam
Eng. @Betree 

## Roadmap (launch ready 28th Sept)
Soft launch (announce only) 8th Sept
Launch ready 28th September

## Prioritization
**Impact**
Collectives will see an increase in contributions through the HAcktoberfest campaign.

**Alignment**
This feature aligns with our current strategy of consolidating down on the existing feature set. While it is mostly of benefit to Open Source Collective the campagin will be open to all OS projects across all hosts. 

**Effort**
There is very little novel development required to support this project. 

## Kick-off meeting

Present @benjam @Betree 

### Proposed Solution ([project board](https://github.com/opencollective/opencollective/projects/67))

**TLDR** Hackertoberfest will use the embed flow. Open Collective will add a tag param to tag contributions made as part of the campaign. Open Collective will provide a search API for participating projects tagged with hacktoberfest. Open Collective will also provide an API with sum of contributions tagged with the API and a list of users who have donated >$20 USD who qualify for a tee shirt. 

- [ ] Projects can opt into the campaign by adding the hacktoberfest tag to their project and enabling the default contribution tier (used in the embedded contribution flow) 
- [ ] Open Collective to provide an API endpoint to return participating projects — note that this API should only return active, tagged projects with a host that has paypal/stripe connection (see risks) https://github.com/opencollective/opencollective/issues/4630
- Hacktoberfest will periodically hit the search API to request a list of participating projects
- Hacktoberfest will use this to present participating projects alongside projects participating in GitHub sponsors
- Hacktoberfest will disambiguate between projects on Open Collective also participating in Sponsors  
- Hacktoberfest will provide free text search functionality on their site
- [ ] Open Collective will add a param for Hacktoberfest to tag contributions made during hacktoberfest. https://github.com/opencollective/opencollective/issues/4626
- [ ] Open Collective will add a note to the embed flow if the project has additional tiers 
- [ ] Open Collective will disable bank transfers as a payment option in the embed during the campaign
- Hacktoberfest will use the embed flow, to allow users to donate to projects including the tag, and user email address (to prefill form and match if user is registered our side)
- [ ] Open Collective will provide an API to return the total sum contributed during the campaign https://github.com/opencollective/opencollective/issues/4631
- [ ] ~~After the campaign Open Collective will provide a list of the email addresses (provided to us by Hacktoberfest when the registered user uses the contribution flow) every contributor who donated $>20 USD~~
- ~~Assumed Hacktoberfest manage address collection as part of user registration~~
- ~~Hacktoberfest will send contributors their tee-shirts~~ 

### Questions, opportunities, further work, and extensions

- What data needs to be returned in the search API? We assume name, slug, description, logo. 
- Do Hacktober fest also want a transaction history to show on their site?
- We _could_ (if we wanted to invest the effort) build multi-project checkout process a la FundOSS. This would allow users to hit the $20 aggregate without having to go through the flow multiple times.
- In future years we could also issue gift cards instead of tee shirts through Open Collective. We'll pick this up next year. 

### Roadmap 
**Soft launch (announce only) 8th Sept
**Launch ready 28th September

### Success metrics
Increase in contributions to participating projects (from current baseline)
Increase in subscribers to participating projects (from baseline)

### Risks
- Risk: CC abuse due to increased attention and access
- Mitigation: add recaptcha to embed form if we are under attack 

- Risk: traffic to image servers likely to increase as hacktoberfest serves up participating projects
- Mitigation: bump up dynos on image service

- Risk: search traffic from hacktoberfest could slow down application performance 
- Mitigation: create new/amend endpoint to provide only participating projects, cache results, ask hacktoberfest to serve search from their side. 

- Risk: abuse vector for folks adding a self-hosted collective and tagging hacktoberfest. 
- Mitigation: only return hosted collectives with a valid stripe/paypal account linked in API. 

- Risk: bank transfer spam could cause the campaign to appear more successful than it really is and place an additional load on support staff to line up transfers
- Mitigation: disable bank transfer in the embed flow, monitor for feedback and reopen if needed 

- Risk: GDPR data sharing agreement doesn't include this
- Mitigation: @benjam will check this, but if we're only returning information (email addresses of participating contributors) back to a provider we received we should be okay. 

### Release planning
- Which aspect(s) of the platform does this change impact? Contribution flow
- Which group(s) of users will benefit from this change? Partners, contributors
- Is this project in response to a user-requested feature? If so who requested it (link to feature request if available)? Hacktoberfest 2021 campaign 
- Is this project related to any similar work we have done, or are planning on doing in the next three months? FundOSS may benefit. 

### Next steps

- [ ] Open up any appropriate project issues
- [ ] Open a documentation issue, link it to this one
- [ ] Open a release issue, link it to this one
- [ ] Create a project board, add this issue to the description, add any issues created above 

## Project Closure

- Close all relevant project issues
- Open up issues for improvements 
- Post comm's according to comm's plan (most likely in release issue)
- Post project screencast on documentation slack channel 
- Post project closure in newsletter slack channel 
- Celebrate!!

**Comment by BenJam** [comment]
Updated to remove the tee shirt process after confirming with @pheobyquincy that they're not planning on doing this for financial contributors.

**Comment by BenJam** [comment]
closed as campaign is over for this year. Retro organised shortly but the figures from our side are poor ($30 raised). If GH Sponsors' figures are considerably different then we will investigate for issues with the embed but otherwise we look forward to expanding next year.
